.github/workflows/daily_deps_test.yml

```diff
@@ -0,0 +1,63 @@
+name: Daily CI Job
+
+on:
+  schedule:
+    - cron: '0 12 * * *' # Daily at midnight UTC
+  pull_request: {} # temporary: run on PRs so we can test this workflow
+
+  # Can be triggered manually from the actions tab, if needed
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: test on Python ${{ matrix.python-version }} and pydantic ${{ matrix.pydantic-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        pydantic-version: ['main']
+        include:
+          - python-version: '3.12'
+            pydantic-version: '2.4'
+          - python-version: '3.12'
+            pydantic-version: '2.5'
+          - python-version: '3.12'
+            pydantic-version: '2.6'
+          - python-version: '3.12'
+            pydantic-version: '2.7'
+          - python-version: '3.12'
+            pydantic-version: '2.8'
+          - python-version: '3.12'
+            pydantic-version: '2.9'
```

Comment:
Why do we do this only with Pydantic?

Reply:
I think we will want a matrix for other deps too but it doesn't need to be a first step. We're still using the latest version of everything.

I also think that having this full matrix here means that regular CI only needs 2.4 and the latest release (pinned) and anything else needed for coverage (eventually v1 when some form of that works) so that it can run quickly and cheaply. Not a blocker though.

Reply:
@alexmojaki,

Wouldn't it make sense to just test `2.4` and `main` on the daily CI, and leave all of these for the PR tests?

Reply:
No, the daily CI is a good place to put things that take longer that we don't want to wait for to merge a PR.